### DISCUSSION TITLE
[Core Protocol] MixHash / PREVRANDAO

### DISCUSSION THREAD

**Discussion by except** [ethereum-optimism/developers#401] [Core Protocol] [answered]
### Did you check the documentation?

- [X] Yes

### Did you read the specs?

- [X] Yes
- [ ] No

### Did you check for duplicate questions?

- [X] Yes

### Issue Description

I see in the docs that this value is pseudorandomly generated, however I cannot seem to find how. Could you please explain or point to the wherever code this is generated? 

Thank you

**Comment by hash1go** [comment]
Hello. Could you please share the link to the document you read?

**Comment by except** [comment]
It states the difference here but does not expand on it.https://docs.optimism.io/chain/differences
On 14 May 2024, at 12:36, hashigo ***@***.***> wrote:﻿
Hello. Could you please share the link to the document you read?

**Reply by hash1go** [reply, marked as answer]
According to [this document](https://specs.optimism.io/protocol/exec-engine.html#ecotone-beacon-block-root), the value of PREVRANDAO appears to be based on the contents of the L1 beacon block root.
